main_pipeline.py

```diff
@@ -28,33 +28,36 @@ def get_video_info(video_path):
     duration = frame_count / fps
     return vr, duration, fps
 
-def calculate_keyframes(video_duration, frame_rate, start_time, end_time):
+def calculate_keyframes(video_duration, frame_rate, start_time, end_time, visual_detail="high"):
```

Comment:
can we restructure this function. super confused by all these constants and random if statements. make it more understandable.

Reply:
there was only one more if statement added and some more constants that just represent the locations of the frames. will write them separately so it's easier to read